contracts/limitOrder/LimitOrderBook.sol

```diff
@@ -219,4 +228,66 @@ contract LimitOrderBook is
 
         return signer;
     }
+
+    //
+    // INTERNAL VIEW
+    //
+
+    function _formatDecimals(
+        uint256 price,
+        uint8 fromDecimals,
+        uint8 toDecimals
+    ) internal pure returns (uint256) {
+        // LOB_ID: Invalid Decimals
+        require(fromDecimals <= toDecimals, "LOB_ID");
+
+        return price.mul(10**(toDecimals.sub(fromDecimals)));
+    }
+
+    function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
+        ChainlinkPriceFeed chainlinkPriceFeed = ChainlinkPriceFeed(IBaseToken(baseToken).getPriceFeed());
+        (uint256 price, ) = chainlinkPriceFeed.getRoundData(roundId);
+        return _formatDecimals(price, chainlinkPriceFeed.decimals(), 18);
+    }
+
+    function _verifyTriggerPrice(LimitOrder memory order, uint80 roundIdWhenTriggered) internal view {
+        // NOTE: Chainlink proxy's roundId is always increased
+        // https://docs.chain.link/docs/historical-price-data/
+
+        // LOB_IRI: Invalid RoundId
+        require(order.roundIdWhenCreated > 0 && roundIdWhenTriggered > order.roundIdWhenCreated, "LOB_IRI");
+
+        // LOB_ITP: Invalid Trigger Price
+        require(order.triggerPrice > 0, "LOB_ITP");
+
+        // NOTE: we can only support stop/take-profit limit order for markets that use ChainlinkPriceFeed
+        uint256 triggeredPrice = _getPriceByRoundId(order.baseToken, roundIdWhenTriggered);
+
+        // we need to make sure the price has reached trigger price.
```

Comment:
Not sure if this issue is fully discussed and I have no idea which price is used to be compared to `triggerPrice` on other exchanges

Reply:
There will be 2 phases to fill a stop-loss/take-profit limit order:

Phase 1:

The [keeper](https://github.com/perpetual-protocol/perp-curie-limit-order/tree/main/packages/keeper) will keep tracking index price changes (through Chainlink aggregator events), and compare the index price with limit orders' trigger price. If index price matches trigger price (the logic should be similar to [this](https://github.com/perpetual-protocol/perp-curie-periphery-contract/blob/feature/stop-limit-order/contracts/limitOrder/LimitOrderBook.sol#L255) I suppose), the keeper will save the index price's `roundId` into `roundIdWhenTriggered` field in DB.

Phase 2:

The keeper fetches limit orders with `roundIdWhenTriggered != 0`, try calling `LimitOrderBook.fillLimitOrder(order, roundIdWhenTriggered)` from time to time.